### PR TITLE
data-sync

### PR DESCRIPTION
**Pull request by wangyue168git**

### CODE REVIEW

File: fisco-bcos/main/CMakeLists.txt [hunk 1]
```diff
@@ -28,9 +28,9 @@ target_link_libraries(fisco-bcos PUBLIC initializer)
 
 install(TARGETS fisco-bcos DESTINATION bin)
 
-if (TOOL)
-    add_executable(fisco-sync DataSyncTool.cpp)
-    target_include_directories(fisco-sync PUBLIC ..)
-    target_link_libraries(fisco-sync PUBLIC initializer storage)
-    install(TARGETS fisco-sync DESTINATION bin)
-endif()
+#if (TOOL)
```
Comment (bxq2011hust) [inline fisco-bcos/main/CMakeLists.txt:31]: it is ok to delete these

File: fisco-bcos/datasync/DataSyncTool.cpp [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+            auto entry = std::make_shared<Entry>();
+            entry->copyFrom(entries->get(i));
+            tableData->dirtyEntries->addEntry(entry);
+        }
+        tableData->newEntries = std::make_shared<Entries>();
+        return tableData;
+    }
+}
+
+void conversionData(const std::string& tableName, TableData::Ptr tableData)
```
Comment (dalaocu) [inline fisco-bcos/datasync/DataSyncTool.cpp:229]: converseData

File: fisco-bcos/datasync/DataSyncTool.cpp [hunk 3]
```diff
@@ -324,51 +328,52 @@ void syncData(SQLStorage::Ptr _reader, Storage::Ptr _writer, int64_t _blockNumbe
...
-
-            if (tableData->newEntries->size() < counts)
-            {
-                break;
-            }
-        }
-        recorder->markStatus(tableInfo->name, make_pair(start, true));
-        ++syncedCount;
-        cout << " done.\r" << flush;
+      cout << endl
```
Comment (bxq2011hust) [inline fisco-bcos/datasync/DataSyncTool.cpp:334]: use `clang-format -i` to format code

File: fisco-bcos/datasync/DataSyncTool.cpp [hunk 4]
```diff
@@ -430,50 +435,40 @@ void syncData(SQLStorage::Ptr _reader, Storage::Ptr _writer, int64_t _blockNumbe
     }
 }
 
-void fastSyncGroupData(std::shared_ptr<LedgerParamInterface> _param,
-    ChannelRPCServer::Ptr _channelRPCServer, int64_t _rollbackNumber = 1000)
+
+void fastSyncData(std::shared_ptr<LedgerParamInterface> _param, std::shared_ptr<LedgerParamInterface> stashParam, uint64_t _stopBlockNumber = 0)
 {
-    if (g_BCOSConfig.version() < V2_6_0)
```
Comment (bxq2011hust) [inline fisco-bcos/datasync/DataSyncTool.cpp:436]: I think we can keep this check?

File: fisco-bcos/datasync/DataSyncTool.cpp [hunk 5]
```diff
@@ -487,42 +482,40 @@ void fastSyncGroupData(std::shared_ptr<LedgerParamInterface> _param,
...
+        auto archiveStorage = rocksDBStorageFactory->getStorage(to_string(_stopBlockNumber));
+        scalableStorage->setArchiveStorage(archiveStorage, _stopBlockNumber);
+        scalableStorage->setRemoteBlockNumber(_stopBlockNumber);
         writerStorage = scalableStorage;
     }
 
     // fast sync data
-    syncData(
-        dynamic_pointer_cast<SQLStorage>(sqlStorage), writerStorage, blockNumber, _param, fullSync);
+    syncData_Link(dynamic_pointer_cast<StashStorage>(readerStorage), writerStorage, _stopBlockNumber, _param, fullSync);
```
Comment (bxq2011hust) [inline fisco-bcos/datasync/DataSyncTool.cpp:492]:
```suggestion
    syncDataLink(dynamic_pointer_cast<StashStorage>(readerStorage), writerStorage, _stopBlockNumber, _param, fullSync);
```

File: fisco-bcos/datasync/DataSyncTool.cpp [hunk 6]
```diff
@@ -487,42 +482,40 @@ void fastSyncGroupData(std::shared_ptr<LedgerParamInterface> _param,
...
 {
-    /// set LC_ALL
-    setDefaultOrCLocale();
-    std::set_terminate([]() {
-        std::cerr << "terminate handler called" << endl;
-        abort();
-    });
-    std::cout << "fisco-sync version : "
-              << "0.1.0" << std::endl;
+    std::cout << "fisco-sync version : " << "0.1.0" << std::endl;
```
Comment (bxq2011hust) [inline fisco-bcos/datasync/DataSyncTool.cpp:499]:
```suggestion
    std::cout << "fisco-sync version : " << "0.2.0" << std::endl;
```

**Review by bxq2011hust — APPROVED** [review]

**Pull request closed without merging**